frontend/src/hooks/useAuth/context.tsx

```diff
@@ -307,7 +307,11 @@ export const AuthProvider: React.FC<IProps> = ({ children }) => {
   const createRole = async (name: string): Promise<boolean> => {
     setCreatingRole(true)
     try {
-      const response = await http.post(`role`, { name: name })
+      const user = Authentication.getUser()
+      const response = await http.post(`role`, {
+        name: name,
+        created_by: user?.id ? Number(user.id) : null,
```

Comment:
Is it possible that there wouldn't be a user authenticated? I am not familiar with the data model in the backend, but is it acceptable for the `created_by` to be null?
If not, would it be better to maybe throw an "unauthorized error" here if `user == null`?